target/riscv/cheri-archspecific.h

```diff
@@ -52,30 +52,15 @@ static inline void QEMU_NORETURN raise_cheri_exception_impl(
     riscv_raise_exception(env, RISCV_EXCP_CHERI, hostpc);
 }
 
-static inline bool
-cheri_tag_prot_clear_or_trap(CPUArchState *env, target_ulong va,
-                             int cb, const cap_register_t* cbp,
-                             int prot, uintptr_t retpc, target_ulong tag)
+static inline void QEMU_NORETURN raise_load_tag_exception(
+    CPUArchState *env, target_ulong va, int cb, uintptr_t retpc)
 {
-#ifndef TARGET_RISCV32
-    if (tag && (prot & PAGE_LC_CLEAR)) {
-        qemu_maybe_log_instr_extra(env, "Clearing tag loaded from " TARGET_FMT_lx
-            " due to missing PTE_LC\n", va);
-        return 0;
-    }
-#endif
-    if (tag && !(cbp->cr_perms & CAP_PERM_LOAD_CAP)) {
-        qemu_maybe_log_instr_extra(env, "Clearing tag loaded from " TARGET_FMT_lx
-            " due to missing CAP_PERM_LOAD_CAP\n", va);
-        return 0;
-    }
-#ifndef TARGET_RISCV32
-    if (tag && (prot & PAGE_LC_TRAP)) {
-      env->badaddr = va;
-      riscv_raise_exception(env, RISCV_EXCP_LOAD_CAP_PAGE_FAULT, retpc);
-    }
+#ifdef TARGET_RISCV32
+    __builtin_unreachable();
```

Comment:
I know this can't currently happen but we may want a cheri_debug_assert() to ensure we don't accidentally break something when we add qemu RISCV 32 support?

Reply:
Sounds good.  I wasn't sure if there a more specific thing to use here than this.

Actually, looks like QEMU has a `g_assert_not_reached()` that might be the thing to use here?